app/src/main/java/org/openobservatory/ooniprobe/activity/runtests/adapter/RunTestsExpandableListViewAdapter.kt

```diff
@@ -0,0 +1,229 @@
+import org.openobservatory.ooniprobe.activity.runtests.RunTestsViewModel.Companion.SELECT_NONE
+import org.openobservatory.ooniprobe.activity.runtests.RunTestsViewModel.Companion.SELECT_ALL
+import org.openobservatory.ooniprobe.activity.runtests.RunTestsViewModel.Companion.SELECT_SOME
+import org.openobservatory.ooniprobe.activity.runtests.models.ChildItem
+import org.openobservatory.ooniprobe.activity.runtests.models.GroupItem
+import org.openobservatory.ooniprobe.test.suite.AbstractSuite
+import org.openobservatory.ooniprobe.test.suite.ExperimentalSuite
+import org.openobservatory.ooniprobe.test.test.AbstractTest
+
+
+class RunTestsExpandableListViewAdapter(
+	private val mContext: Context,
+	private val mGroupListData: List<GroupItem>,
+	private val mViewModel: RunTestsViewModel
+) : BaseExpandableListAdapter() {
+	override fun getGroupCount(): Int {
+		return mGroupListData.size
+	}
+
+	override fun getChildrenCount(groupPosition: Int): Int = mGroupListData[groupPosition].nettests.size
+
+	override fun getGroup(groupPosition: Int): GroupItem = mGroupListData[groupPosition]
+
+	override fun getChild(groupPosition: Int, childPosition: Int): ChildItem =
+		mGroupListData[groupPosition].nettests[childPosition]
+
+	override fun getGroupId(groupPosition: Int): Long = groupPosition.toLong()
+
+	override fun getChildId(groupPosition: Int, childPosition: Int): Long = childPosition.toLong()
+
+	override fun hasStableIds(): Boolean = false
+
+	override fun getGroupView(groupPosition: Int, isExpanded: Boolean, convertView: View?, parent: ViewGroup): View? {
+		var convertView =
+			convertView ?: LayoutInflater.from(mContext).inflate(R.layout.run_tests_group_list_item, parent, false)
+		val groupItem = getGroup(groupPosition)
+		val testSuite = AbstractSuite.getTestSuiteByName(groupItem.name)
+		convertView.findViewById<TextView>(R.id.group_name).text = parent.context.resources.getText(testSuite.title)
+		convertView.findViewById<ImageView>(R.id.group_icon).setImageResource(testSuite.icon)
+		val groupIndicator = convertView.findViewById<ImageView>(R.id.group_indicator)
+		val groupSelectionIndicator = convertView.findViewById<ImageView>(R.id.group_select_indicator)
+		val selectedAllBtnStatus = mViewModel.selectedAllBtnStatus.getValue()
+		if (selectedAllBtnStatus == SELECT_ALL) {
+			groupItem.selected = true
+			for (childItem in groupItem.nettests) {
+				childItem.selected = true
+			}
+		} else if (selectedAllBtnStatus == SELECT_NONE) {
+			groupItem.selected = false
+			for (childItem in groupItem.nettests) {
+				childItem.selected = false
+			}
+		} else if (isSelectAllChildItems(groupItem.nettests)) {
+			groupItem.selected = true
+		}
+		if (groupItem.selected) {
+			if (isSelectAllChildItems(groupItem.nettests)) {
+				groupSelectionIndicator.setImageResource(R.drawable.check_box)
+				// NOTE: This is the only place where ExperimentalSuite.NAME is used.
+				// This doesn't follow the normal rule where the component tests make up the suite.
```

Comment:
Could you elaborate a bit more in the comment in what needs to happen to avoid using the `.NAME` explicitly?

For context: I am happy that we documented this limitation, but I'm still a bit in the dark with respect to understanding why this is a limitation and about the next steps to remove it.

Reply:
Experimental.NAME is used as a key to identify the experimental test suite.
Here it is treated differently since unlike the suites that depend on the component parts to be marked as checked or not, the experimental preference determines if the components are enabled or not